std/mem.zig

```diff
@@ -57,8 +57,10 @@ pub const Allocator = struct {
 
     /// `ptr` should be the return value of `create`
     pub fn destroy(self: *Allocator, ptr: var) void {
-        const non_const_ptr = @intToPtr([*]u8, @ptrToInt(ptr));
-        self.freeFn(self, non_const_ptr[0..@sizeOf(@typeOf(ptr).Child)]);
+        const bytes = @sliceToBytes(memory);
```

Comment:
I don't think this change is correct. `destroy` vs ` free` on, for example, a slice type is a fundamentally different operation.